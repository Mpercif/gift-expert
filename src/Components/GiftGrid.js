import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../Hooks/useFetchGifs'
import GiftGridItem from './GiftGridItem'

const GiftGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h2>{category}</h2>
        {loading && <p className="animate__bounce">'Cargando...'</p>  }
        
        <div className='card-grid animate__bounce' > 
            {
                images.map(img => 
                    <GiftGridItem 
                        key={img.id}
                        {...img} 
                    /> 
                )
            }
        </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GiftGrid
