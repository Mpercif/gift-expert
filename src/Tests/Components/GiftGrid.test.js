import { shallow } from "enzyme"
import '@testing-library/jest-dom'
import React from 'react';
import GiftGrid from "../../Components/GiftGrid"
import { useFetchGifs } from "../../Hooks/useFetchGifs";

jest.mock("../../Hooks/useFetchGifs")


describe('Pruebas del componente GifGrid', () => {
    const category = 'Dragon Ball'
       
    test('Renderizacion del componente ', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        } )

        const wraper = shallow(<GiftGrid category={category} /> )
        expect(wraper).toMatchSnapshot()
    })

    test('Renderizacion del componente ', () => {

        const imgs = [{
            id: '123467432',
            title: 'ABCDES',
            url: 'https://localhost.com/cualquier'
        },
        {
            id: '8465416546',
            title: 'ABCDES',
            url: 'https://localhost.com/cualquier'
        }]

        useFetchGifs.mockReturnValue({
            data:imgs,
            loading: false
        } )

        const wraper = shallow(<GiftGrid category={category} /> )

               
        expect(wraper.find('p').exists()).toBe(false)
        expect(wraper.find('GiftGridItem').length).toBe(imgs.length)
    })

    
    
})
