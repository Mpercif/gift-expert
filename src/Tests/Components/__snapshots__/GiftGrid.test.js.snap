// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid Renderizacion del componente  1`] = `
<Fragment>
  <h2>
    Dragon Ball
  </h2>
  <p
    className="animate__bounce"
  >
    'Cargando...'
  </p>
  <div
    className="card-grid animate__bounce"
  />
</Fragment>
`;
